src/nunit.analyzers/SameAsOnValueTypes/SameAsOnValueTypesCodeFix.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NUnit.Analyzers.Constants;
+
+namespace NUnit.Analyzers.SameAsOnValueTypes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    [Shared]
+    public class SameAsOnValueTypesCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(AnalyzerIdentifiers.SameAsOnValueTypes);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var diagnostic = context.Diagnostics.First();
+            var invocationNode = root.FindNode(diagnostic.Location.SourceSpan) as InvocationExpressionSyntax;
+
+            if (invocationNode == null)
+                return;
+
+            // Find the 'SameAs' member access expression
+            var constraintArgument = invocationNode.ArgumentList.Arguments[1];
+            var isSameExpression = constraintArgument.DescendantNodesAndSelf()
+                                                           .Where(s => s.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+                                                           .OfType<MemberAccessExpressionSyntax>()
+                                                           .Where(m => m.Name.ToString() == NunitFrameworkConstants.NameOfIsSameAs)
+                                                           .SingleOrDefault();
+
+            var isEqualToExpression = isSameExpression.WithName(SyntaxFactory.IdentifierName(NunitFrameworkConstants.NameOfIsEqualTo));
```

Comment:
Perhaps we should have a return in case `isSameExpression` is null

Reply:
Done. 
However should this fix not only be called for the supported diagnostic, which would mean there should always be an `SameAs`?  Maybe it is better to be more conservative than throw.

Reply:
I agree that this should never fail, but I plan to enable more strict checks of nullable reference types, and then we will need to handle the case, as the compiler cannot tell that this cannot be null.

Reply:
Will you upgrade to a later version of Microsoft.CodeAnalysis which has nullable attributes? When I tried, it requires me to target netstandard2.0 and after that I got various nullability compile errors (CS) and analyzer violations (RS).

Reply:
I'm considering trying to multi-target, and see if I can make it work, but currently I don't have enough sparetime to do much besides reviewing and screening issues in the various NUnit projects.

So something similar to https://www.meziantou.net/how-to-use-nullable-reference-types-in-dotnet-standard-2-0-and-dotnet-.htm

Reply:
We added multi-targeting to our work code. Create an issue with what targets you want and assign it to me.